### DISCUSSION TITLE
Webhook Subscription

### DISCUSSION THREAD

**Discussion by abdusfauzi** [RichieMcMullen/laravel-strava#27] [Ideas]
Firstly, thank you! Love the package ❤️ 

Next, is it hard to implement webhook subscription API? I think it would be a great addition to the available set of features.
https://developers.strava.com/docs/webhooks/

Thank you.
👍

**Comment by boryn** [comment]
I'd say it was not that difficult to implement Strava webhooks. I used for it this package: https://github.com/spatie/laravel-webhook-client. With it you can easily handle incoming webhooks in the background jobs.

You need to remember about few things:
- Strava's webhook call does not use any secret key (known to you and them) and does send send any hash of the payload so that you could confirm it is really coming from them.
- It's advisable to control api limits. You can check them using this library with `\Strava::getApiLimits()`.
- Take into account that webhook can inform you "now" about some activity from the past (eg. when user changed its name) so carefully look at the dates.
- User can as well change other things, like activity type, etc. so take into account updating the information in your database, not only storing.

For more discussions on Strava API you may join https://groups.google.com/g/strava-api

**Comment by breadthe** [comment]
Webhooks would be great. I'm in the process of implementing them on a Strava Laravel app, outside of this package.

I've contributed here in the past so there's a chance I will submit a PR at some point. I can't promise anything though, especially since the implementation will be an MVP of sorts, good enough for me but not for others. I'm not going to bother covering all the edge cases or writing tests, not initially anyway.

**Comment by breadthe** [comment]
Alright, so I built webhook functionality into my app over the weekend.

Here's all the relevant code, without context: https://gist.github.com/breadthe/2787c4f6d6ac805ef9eb698a91b6a750

If you want a more in-depth discussion of how & why, I also wrote a blogpost: https://chasingcode.dev/blog/laravel-strava-webhook-solution/
